docs/Tutorial_DiscordMERLIN.ipynb

```diff
@@ -0,0 +1,1780 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #73.                # stop + m - 1 is correct for converting stop...
</pre><p>To avoid this comment, maybe we can change <code>_get_chunks_ranges(non_trivial_cands_idx)</code> so that it returns the <code>start</code> and <code>stop</code> ranges relative to <code>T</code> rather than relative to the subsequences.</p><p> </p><pre class="ql-syntax" spellcheck="false">def _get_chunks_ranges(a, shift=None):
</pre><p> </p><p>Then, at the end of <code>_get_chunks_ranges</code> :</p><pre class="ql-syntax" spellcheck="false">if shift is not None:
    out[:, 1] += shift  
</pre><p> </p><p>And then you'd call the function with:</p><pre class="ql-syntax" spellcheck="false">cand_idx_chunks = _get_chunks_ranges(non_trivial_cands_idx, shift=m-1)
for start, stop in cand_idx_chunks:
    QT = core.sliding_dot_product(T[i:i+m], T[start: stop])&nbsp;&nbsp; &nbsp; &nbsp; &nbsp;&nbsp;
&nbsp; &nbsp; D = core._mass(T[i:i+m], T[start:stop], QT, M_T[i], Σ_T[i], M_T[start:stop-m+1], Σ_T[start:stop-m+1])
    mask = np.flatnonzero(D &lt; min_dist)&nbsp; &nbsp;
&nbsp; &nbsp; is_cands[start:stop-m+1][mask] = False
</pre><p>I think this is more natural/obvious and allows us to avoid future confusion</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/TDAmeritrade/stumpy/pull/505/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

Reply:
Sounds good! I may just add a small comment on `shift = m - 1` to help user understand why it is not `m`. I will change it.

Reply:
<p>Sounds good</p>

Reply:
@seanlaw 
Instead of:
```
if shift is not None:
    out[:, 1] += shift
```
I did:
```
if shift is None:
    shift = 1
```
And, then at the end, I changed `out[:, 1] += 1` to `out[:, 1] += shift`. 
(If you think it is making it more complicated in terms of readability, please let me know and I will change and push again)

(Btw, we can just default `shift` to 1 rather than `None` and we can simply get rid of `if shift is None` check)

Reply:
<p>@NimaSarajpoor I don't like it for multiple reasons. In addition to poorer readability, you have put an unnecessary constraint on <code>shift</code> needing to be greater than 1. I think this function is less general now and less flexible and it makes it harder for other functions to use this in the future. Additionally, I actually prefer <code>shift=m-1</code> because it is closely related to converting from time series space to subsequence space. When you change this to <code>shift=m</code> , it is no longer obvious. Please revert this function back.  </p>

Reply:
Thanks for the explanation. I changed it back.